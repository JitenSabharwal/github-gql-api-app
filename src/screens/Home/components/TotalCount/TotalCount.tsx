import React from "react"
import styled from "styled-components"

const Root = styled.div`
  display: flex;
  width: 100%;
  margin-top: 8px;
  font-size: 13px;
  line-height: 16px;
`
const Text = styled.span`
  margin-right: 8px;
`
const Count = styled.span`
  color: #6c757d;
`
export default function TotalCount({
  value,
  label,
}: {
  value: number
  label: string
}) {
  return (
    <Root>
      <Text>{label}</Text>
      <Count>{value}</Count>
    </Root>
  )
}
