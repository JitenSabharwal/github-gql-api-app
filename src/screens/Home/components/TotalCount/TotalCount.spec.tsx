import TotalCount from "./TotalCount"
import React from "react"
import { shallow } from "enzyme"

describe("TotalCount", () => {
  let container: React.ReactElement
  beforeAll(
    () =>
      (container = (
        <TotalCount value={100} label={"Total number of found issues"} />
      ))
  )
  it("Should render the compoent", () => {
    const component = shallow(container)
  })
  it("Should render the layout", () => {
    const component = shallow(container)
    expect(component.getElements()).toMatchSnapshot()
  })
})
