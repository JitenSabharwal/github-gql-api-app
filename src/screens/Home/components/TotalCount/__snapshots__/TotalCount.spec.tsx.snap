// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalCount Should render the layout 1`] = `
Array [
  <styled.div>
    <styled.span>
      Total number of found issues
    </styled.span>
    <styled.span>
      100
    </styled.span>
  </styled.div>,
]
`;
